habitat-hitl/habitat_hitl/_internal/networking/networking_process.py

```diff
@@ -257,29 +228,83 @@ async def serve(self, websocket):
             self.handle_disconnect()
 
 
-def networking_main(interprocess_record, exit_event):
+async def start_websocket_server(network_mgr, networking_config):
     global use_ssl
+    network_mgr_lambda = lambda ws, path: network_mgr.handle_connection(ws)
+    ssl_context = create_ssl_context() if use_ssl else None
+    websocket_server = await websockets.serve(
+        network_mgr_lambda, "0.0.0.0", networking_config.port, ssl=ssl_context
+    )
+    print(
+        f"NetworkManager started on networking thread. Listening for client websocket connections on port {networking_config.port}..."
+    )
+    return websocket_server
+
+
+async def start_http_availability_server(network_mgr, networking_config):
+    async def http_handler(request):
+        # return an HTTP code to indicate available or not
+        code = (
+            networking_config.http_availability_server.code_unavailable
+            if network_mgr.has_connection()
+            else networking_config.http_availability_server.code_available
+        )
+        print(f"Returned availability HTTP code {code}")
+        return aiohttp.web.Response(status=code)
+
+    app = aiohttp.web.Application()
+    app.router.add_get("/", http_handler)
+    runner = aiohttp.web.AppRunner(app)
+    await runner.setup()
+    site = aiohttp.web.TCPSite(
+        runner, "0.0.0.0", networking_config.http_availability_server.port
+    )
+    await site.start()
+    print(
+        f"HTTP availability server started on networking thread. Listening for HTTP requests on port {networking_config.http_availability_server.port}..."
+    )
 
-    if multiprocessing_config.use_dummy:
-        loop = asyncio.new_event_loop()
-        asyncio.set_event_loop(loop)
+    return runner
 
-    network_mgr = NetworkManager(interprocess_record, exit_event)
-    network_mgr_lambda = lambda ws, path: network_mgr.serve(ws)
-    ssl_context = create_ssl_context() if use_ssl else None
-    start_server = websockets.serve(
-        network_mgr_lambda, "0.0.0.0", 8888, ssl=ssl_context
+
+async def networking_main_async(interprocess_record):
```

Comment:
The big refactor here is that all our networking logic is now implemented with `async/await`. Reviewers should take some time to make sure they understand how `networking_main_async` and `networking_main` work.